fuzz/fuzz_security_deser/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+project(fuzz_security_deser LANGUAGES C)
+cmake_minimum_required(VERSION 3.5)
+
+if(NOT TARGET CycloneDDS::ddsc)
+  # Find the CycloneDDS package.
+  find_package(CycloneDDS REQUIRED)
+endif()
+
+add_executable(fuzz_security_deser fuzz_security_deser.c)
+target_include_directories(
+  fuzz_security_deser PRIVATE
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../src/core/ddsi/include>"
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../src/security/core/include>"
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../src/security/api/include>")
+target_link_libraries(fuzz_security_deser CycloneDDS::ddsc $ENV{LIB_FUZZING_ENGINE})
```

Comment:
I learnt the hard way that a variety of fuzzing engines is supported and used by the oss-fuzz infrastructure — at least "libfuzzer", "afl", "honggfuzz" and "centipede". "Centipede" appears to require linking in C++ code, and the easiest way to do that is to set the linker language to C++. I don't think it is necessary at this time to prevent build error reports from oss-fuzz, but it is quite a trivial change to make.

```suggestion
set_target_properties(fuzz_security_deser PROPERTIES LINKER_LANGUAGE CXX)
target_link_libraries(fuzz_security_deser CycloneDDS::ddsc $ENV{LIB_FUZZING_ENGINE})
```

Reply:
I have to admit, I only tried libFuzzer. Added the suggest in a separate commit.